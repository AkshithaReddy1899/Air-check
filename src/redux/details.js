import { v4 as uuidv4 } from 'uuid';

const GET_DETAILS = 'apiPollution/GET_DATA';

const initialState = [];

export const getData = (arr) => ({
  type: GET_DETAILS,
  payload: arr,
});

export const getDetailsFromAPI = (param) => async (dispatch) => {
  console.log(param);
  await fetch(`https://restcountries.com/v3.1/name/${param}`, {
    method: 'GET',
  })
    .then((res) => res.json())
    .catch((error) => {
      const errorText = document.getElementsByClassName('error');
      errorText.textContent = error;
    })
    .then((data) => {
      Object.entries(data).forEach(([key, value]) => {
        const item = {};
        item.num = key;
        item.id = uuidv4();
        item.flag = value.flags.svg;
        item.name = value.name.common;
        item.official = value.name.official;
        item.pop = value.population;
        item.area = value.area;
        item.cap = value.capital;
        item.car = value.car.side;
        item.continent = value.continent;
        item.region = value.region;
        item.subregion = value.subregion;
        dispatch(getData(item));
      });
    });
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_DETAILS:
      return [...state, action.payload];
    default:
      return state;
  }
};

export default reducer;
