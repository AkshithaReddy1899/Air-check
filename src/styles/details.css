.flag {
  width: 100px;
  height: auto;
}

.details {
  text-align: center;
}

.details-main {
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-around;
  align-items: center;
  margin: 40px auto 20px auto;
}

.details-div {
  margin: 10px auto;
  padding: 5px;
  float: right;
  width: 100vw;
  background-color: rgb(81, 81, 228);
}

.detail {
  display: flex;
  justify-content: space-between;
  padding: 10px 15px;
}

.odd {
  background-color: rgb(44, 44, 185);
}