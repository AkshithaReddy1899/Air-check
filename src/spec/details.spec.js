import React from 'react';
import { Provider } from 'react-redux';
import matches from './test-utils';
import store from '../redux/configStore';
import details from '../redux/details';

describe('details redux', () => {
  it('details matched snapshot', () => {
    matches(<Provider store={store}><details /></Provider>);
  });
});
