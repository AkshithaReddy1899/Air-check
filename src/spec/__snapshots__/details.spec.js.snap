// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Homepage matched snapshot 1`] = `<details />`;

exports[`details redux details matched snapshot 1`] = `<details />`;
